src/Microsoft.TemplateEngine.Edge/Template/TemplateCreator.cs

```diff
@@ -103,7 +103,9 @@ public TemplateCreator(IEngineEnvironmentSettings environmentSettings)
                 return new TemplateCreationResult(CreationResultStatus.NotFound, templateInfo.Name, LocalizableStrings.TemplateCreator_TemplateCreationResult_Error_CouldNotLoadTemplate);
             }
 
-            string? realName = name ?? fallbackName ?? template.DefaultName;
+            string? realName = name is not null ? name : templateInfo.PreferDefaultName ? templateInfo.DefaultName : fallbackName;
+
+            // TODO: add template validation here
```

Comment:
Is this still valid?

Reply:
Which part do you mean? 
 - The TODO was there to remind me to do template name validation, but I decided it was out of scope for the current PR if we wanted to change how naming works.
 - If it is the ```if else``` statement, it is to add a new logic so we do not rely first on the fallback name, and use that only as a last option.